packages/react-components/react-progress/src/components/Progress/useProgressStyles.ts

```diff
@@ -1,33 +1,201 @@
+  base: {
+    gridColumnStart: '1',
+    gridRowStart: '2',
+    backgroundColor: tokens.colorCompoundBrandBackground,
+
+    '@media screen and (forced-colors: active)': {
+      backgroundColor: 'Highlight',
+    },
+  },
+  medium: {
+    height: barThicknessValues.medium,
+  },
+  large: {
+    height: barThicknessValues.large,
+  },
+  determinate: {
+    width: `var(${progressCssVars.percentageCssVar})`,
+  },
+  nonZeroDeterminate: {
+    width: `var(${progressCssVars.percentageCssVar})`,
+    transitionProperty: 'width',
+    transitionDuration: '0.3s',
+    transitionTimingFunction: 'ease',
+  },
+  indeterminate: {
+    maxWidth: '33%',
+  },
+  ltr: {
+    position: 'relative',
+    backgroundImage: `linear-gradient(
+      to right,
+      ${tokens.colorNeutralBackground6} 0%,
+      ${tokens.colorCompoundBrandBackground} 50%,
+      ${tokens.colorNeutralBackground6} 100%
+    )`,
+    animationName: IndeterminateProgress,
+    animationDuration: '3s',
+    animationIterationCount: 'infinite',
+  },
+
+  rtl: {
+    animationDirection: 'reverse',
   },
+});
+
+const useTrackStyles = makeStyles({
+  base: {
+    gridRowStart: '2',
+    gridColumnStart: '1',
+    backgroundColor: tokens.colorNeutralBackground6,
 
-  // TODO add additional classes for different states and/or slots
+    '@media screen and (forced-colors: active)': {
+      ...shorthands.borderBottom('1px', 'solid', 'CanvasText'),
+    },
+  },
+  medium: {
+    height: barThicknessValues.medium,
+  },
+  large: {
+    height: barThicknessValues.large,
+  },
 });
 
 /**
  * Apply styling to the Progress slots based on the state
  */
 export const useProgressStyles_unstable = (state: ProgressState): ProgressState => {
-  const styles = useStyles();
-  state.root.className = mergeClasses(progressClassNames.root, styles.root, state.root.className);
+  const { indeterminate = false, thickness = 'medium', percentComplete = 0 } = state;
+  const rootStyles = useRootStyles();
+  const barStyles = useBarStyles();
+  const trackStyles = useTrackStyles();
+  const labelStyles = useLabelStyles();
+  const descriptionStyles = useDescriptionStyles();
+  const { dir } = useFluent();
+
+  const valuePercent = !indeterminate ? Math.min(100, Math.max(0, percentComplete! * 100)) : undefined;
+  const progressBarStyles = {
+    [`${progressCssVars.percentageCssVar}`]: !indeterminate ? valuePercent + '%' : undefined,
+  };
+
+  state.root.className = mergeClasses(progressClassNames.root, rootStyles.root, state.root.className);
+
+  // Determinate indicator when determinate is defined
+  if (state.bar) {
+    state.bar.className = mergeClasses(
+      progressClassNames.bar,
+      barStyles.base,
+      indeterminate && barStyles.indeterminate,
+      indeterminate && barStyles[dir],
+      barStyles[thickness],
+      !indeterminate && barStyles.determinate,
+      !indeterminate && !!valuePercent && valuePercent > ZERO_THRESHOLD && barStyles.nonZeroDeterminate,
```

Comment:
Since `percentComplete` is defaulted above, you could simplify this to
```suggestion
      !indeterminate && percentComplete > ZERO_THRESHOLD && barStyles.nonZeroDeterminate,
```